src/fermilib/utils/_operator_utils.py

```diff
@@ -151,19 +150,17 @@ def save_operator(operator, file_name=None, data_directory=None):
     else:
         raise TypeError('Operator of invalid type.')
 
-    with h5py.File(file_path, 'w') as f:
-        f['operator_type'] = numpy.string_(operator_type)
-        f['operator_terms'] = numpy.string_(json.dumps(
-            [{'k': k, 'r': v.real, 'i': v.imag}
-                for k, v in operator.terms.items()],
-            ensure_ascii=False).encode('utf8'))
+    tm = operator.terms
+    f = open(file_path, 'wb')
+    marshal.dump((operator_type, {k: numpy_scalar(tm[k]) for k in tm}), f)
```

Comment:
After looking at this a little more, I'm not sure numpy_scalar will do anything fundamentally different from casting to complex directly - do you have any thoughts on this @Spaceenter ? Using complex instead is a little cleaner (no extra function needed) and also a bit faster.

By doing that as well as using itertools' izip and imap, this line can be made ~25% faster (`dict(izip(iterkeys(tm), imap(complex, itervalues(tm))))`), though it's by no means a bottleneck :p